src/main/java/io/openliberty/tools/intellij/util/LibertyProjectUtil.java

```diff
@@ -149,10 +150,13 @@ public static ArrayList<BuildFile> getRemovableMavenBuildFiles(Project project)
      * @param createWidget  true if a new widget should be created
      * @return ShellTerminalWidget or null if it does not exist
      */
-    public static ShellTerminalWidget getTerminalWidget(Project project, LibertyModule libertyModule, boolean createWidget, TerminalView terminalView, ShellTerminalWidget widget) {
+    public static ShellTerminalWidget getTerminalWidget(Project project, LibertyModule libertyModule, boolean createWidget,
+                                                        TerminalToolWindowManager terminalToolWindowManager, ShellTerminalWidget widget) {
         if (widget == null && createWidget) {
             // create a new terminal tab
-            ShellTerminalWidget newTerminal = terminalView.createLocalShellWidget(project.getBasePath(), libertyModule.getName(), true);
+            ShellTerminalWidget newTerminal = ShellTerminalWidget.toShellJediTermWidgetOrThrow(
```

Comment:
@aparnamichael , Can you explain the usage of method `toShellJediTermWidgetOrThrow()`?

Reply:
This is the definition of toShellJediTermWidgetOrThrow.
```
public static @NotNull ShellTerminalWidget toShellJediTermWidgetOrThrow(@NotNull TerminalWidget widget) {
    return (ShellTerminalWidget)Objects.requireNonNull(JBTerminalWidget.asJediTermWidget(widget));
  }
```
  
  This method is basically used to typecast the widget to ShellTerminalWidget object. JBTerminalWidget.asJediTermWidge - This method checks whether widget is an instanceof TerminalWidgetBridge. If yes,  it will return bridge.widget() or else returns null. Then the 'requireNonNull' method will check whether 'JBTerminalWidget.asJediTermWidget(widget)' returns null or not. If it is null it will throw an exception or else it will type cast the object to ShellTerminalWidget.
  
When we were using 'createLocalShellWidget' method of TerminalView class. We were doing the same.
  
```
  fun createLocalShellWidget(workingDirectory: String?,
                             tabName: @Nls String?,
                             requestFocus: Boolean): ShellTerminalWidget {
    return ShellTerminalWidget.toShellJediTermWidgetOrThrow(
      toolWindowManager.createShellWidget(workingDirectory, tabName, requestFocus, true))
  }
```